ophyd/log.py

```diff
@@ -100,47 +115,85 @@ def __init__(self, fmt=DEFAULT_FORMAT, datefmt=DEFAULT_DATE_FORMAT,
                 # colorama on windows), assume hard-coded ANSI color codes.
                 for levelno, code in colors.items():
-                    self._colors[levelno] = '\033[2;3%dm' % code
-                self._normal = '\033[0m'
+                    self._colors[levelno] = "\033[2;3%dm" % code
+                self._normal = "\033[0m"
         else:
-            self._normal = ''
+            self._normal = ""
 
     def format(self, record):
-        record.message = record.getMessage()
+        message = []
+        if hasattr(record, "ophyd_object_name"):
+            message.append(f"[{record.ophyd_object_name}]")
+        elif hasattr(record, "status"):
+            message.append(f"[{record.status}]")
+        else:
+            ...
+
+        message.append(record.getMessage())
+        record.message = " ".join(message)
         record.asctime = self.formatTime(record, self.datefmt)
 
         try:
             record.color = self._colors[record.levelno]
             record.end_color = self._normal
         except KeyError:
-            record.color = ''
-            record.end_color = ''
+            record.color = ""
+            record.end_color = ""
 
         formatted = self._fmt % record.__dict__
 
         if record.exc_info and not record.exc_text:
             record.exc_text = self.formatException(record.exc_info)
         if record.exc_text:
-            formatted = '{}\n{}'.format(formatted.rstrip(), record.exc_text)
+            formatted = "{}\n{}".format(formatted.rstrip(), record.exc_text)
         return formatted.replace("\n", "\n    ")
 
 
-plain_log_format = "[%(levelname)1.1s %(asctime)s.%(msecs)03d %(module)s:%(lineno)d] %(message)s"
-color_log_format = ("%(color)s[%(levelname)1.1s %(asctime)s.%(msecs)03d "
-                    "%(module)s:%(lineno)d]%(end_color)s %(message)s")
-logger = logging.getLogger('ophyd')
+plain_log_format = (
+    "[%(levelname)1.1s %(asctime)s.%(msecs)03d %(module)s:%(lineno)d] %(message)s"
+)
+color_log_format = (
+    "%(color)s[%(levelname)1.1s %(asctime)s.%(msecs)03d "
+    "%(module)s:%(lineno)d]%(end_color)s %(message)s"
+)
+
+
+def validate_level(level) -> int:
+    """
+    Return an int for level comparison
+    """
+    if isinstance(level, int):
+        levelno = level
+    elif isinstance(level, str):
+        levelno = logging.getLevelName(level)
+
+    if isinstance(levelno, int):
+        return levelno
+    else:
+        raise ValueError(
+            "Your level is illegal, please use "
+            "'ERROR', 'WARNING', 'INFO', 'DEBUG', OR 'TRACE'."
+        )
+
+
+logger = logging.getLogger("ophyd")
+control_layer_logger = logging.getLogger("ophyd_control_layer")
 
 
 current_handler = None  # overwritten below
 
 
-def set_handler(file=sys.stdout, datefmt='%H:%M:%S', color=True):
+def config_ophyd_logging(
+    file=sys.stdout, datefmt="%H:%M:%S", color=True, level="INFO", backupCount=4
```

Comment:
The default should be `"WARNING"`. That's Python's default level in the logging module, INFO is very verbose (as I just found trying to uses this branch), and it's consistent with https://github.com/caproto/caproto/blob/1a7356487fd930cb6dee6c26f1ce1f117cd87c95/caproto/_log.py#L353.

Reply:
I agree.